### ISSUE TITLE
Empty pdf generated, php 7.3

### ISSUE THREAD

**Issue by tahrz** [dompdf/dompdf#1995]
Hi! I have a problem with pdf generation. File is empty, but HTML is valid...

```php
    $options = new Options();
    $options->setIsRemoteEnabled(true)
        ->setDpi(110)
        ->setIsHtml5ParserEnabled(true)
        ->setIsJavascriptEnabled(true)
        ->setIsPhpEnabled(true)
        ->setIsFontSubsettingEnabled(true);

    $dompdf = (new Dompdf())->setOptions($options);
    $doompdf->loadHtml('<h1>Hello world!</h1>')

    $doompdf->setPaper('A4', 'portrait')->render();

    //then code to save pdf into folder
```

so... It's empty. But if comment `setIsHtml5ParserEnabled` and `setIsJavascriptEnabled` - the situation is normalizing. What is wrong with this two settings?

**Comment by Orange06** [comment]
Hi,
you have a lot of options there so have you tried mayb like this:

`require_once 'dompdf/lib/html5lib/Parser.php';
require_once 'dompdf/lib/php-font-lib/src/FontLib/Autoloader.php';
require_once 'dompdf/lib/php-svg-lib/src/autoload.php';
require_once 'dompdf/src/Autoloader.php';
Dompdf\Autoloader::register();
// reference the Dompdf namespace
use Dompdf\Dompdf;
use Dompdf\Options;

$options = new Options();
$options->set('defaultFont', 'Helvetica');
$dompdf = new Dompdf($options);

// instantiate and use the dompdf class
$pdf = '<h1>Hello</h1>;
$dompdf = new Dompdf();
$dompdf->loadHtml($pdf);
// (Optional) Setup the paper size and orientation
$dompdf->setPaper('A4', 'portrait');
// Render the HTML as PDF
$dompdf->render();
// Output the generated PDF to Browser
$dompdf->stream();`

This example right here works with me, so I think you should start with this and then add you options one by one to trace down the error.
Also, it would help if you would include error reporting in your php ->
`ini_set('display_errors',1);
ini_set('display_startup_errors',1);
ini_set("allow_url_fopen", "on");
error_reporting(-1);`

**Comment by tahrz** [comment]
> Hi,
> you have a lot of options there so have you tried mayb like this:
> 
> `require_once 'dompdf/lib/html5lib/Parser.php';
> require_once 'dompdf/lib/php-font-lib/src/FontLib/Autoloader.php';
> require_once 'dompdf/lib/php-svg-lib/src/autoload.php';
> require_once 'dompdf/src/Autoloader.php';
> Dompdf\Autoloader::register();
> // reference the Dompdf namespace
> use Dompdf\Dompdf;
> use Dompdf\Options;
> 
> $options = new Options();
> $options->set('defaultFont', 'Helvetica');
> $dompdf = new Dompdf($options);
> 
> // instantiate and use the dompdf class
> $pdf = "
> 
> # Hello
> ";
> $dompdf = new Dompdf();
> $dompdf->loadHtml($pdf);
> // (Optional) Setup the paper size and orientation
> $dompdf->setPaper('A4', 'portrait');
> // Render the HTML as PDF
> $dompdf->render();
> // Output the generated PDF to Browser
> $dompdf->stream();`
> This example right here works with me, so I think you should start with this and then add you options one by one to trace down the error.
> Also, it would help if you would include error reporting in your php ->
> `ini_set('display_errors',1); ini_set('display_startup_errors',1); ini_set("allow_url_fopen", "on"); error_reporting(-1);`

In my case, code already works when php was 7.2, after php up to 7.3, pdf makes empty with those two options. From README of Doompdf:

> To avoid any unexpected rendering issues you should either enable the built-in HTML5 parser at runtime ($dompdf->set_option('isHtml5ParserEnabled', true);)

So okay, this option I don't need, in any case I can check template on valid with W3C. But what is wrong with `setIsJavascriptEnabled`?

**Comment by Orange06** [comment]
Im sorry I didnt read carefully your question... when I wrote that before I saw 

> so... It's empty. But if comment setIsHtml5ParserEnabled and setIsJavascriptEnabled - the situation is normalizing. What is wrong with this two settings?

One other thing I've noticed,
Im using $options->set('isHtml5ParserEnabled', true) 
but you are using 
->setIsHtml5ParserEnabled(true)

**Comment by tahrz** [comment]
> Im sorry I didnt read carefully your question... when I wrote that before I saw
> 
> > so... It's empty. But if comment setIsHtml5ParserEnabled and setIsJavascriptEnabled - the situation is normalizing. What is wrong with this two settings?
> 
> One other thing I've noticed,
> Im using $options->set('isHtml5ParserEnabled', true)
> but you are using
> ->setIsHtml5ParserEnabled(true)
> Im not entirely sure but your options is/are working fine on my end but I do have php 5.6 version enabled not 7 so this should also be considered.

But in 7.2 it works :) and https://github.com/dompdf/dompdf/blob/d30679a47a067a69540c988405cb675404898acc/src/Options.php#L345 the way to enable of this option doesn't depend from php version, as I see

**Comment by Orange06** [comment]
I see :) The only thing left is php 5.6 like I said it works here, but if you are strictly bounded to 7+, then my friend I'm sorry and wish you best of luck :)

**Comment by tahrz** [comment]
> I see :) The only thing left is php 5.6 like I said it works here, but if you are strictly bounded to 7+, then my friend I'm sorry and wish you best of luck :)

ohhh... okay, thank you for help)

**Comment by bsweeney** [comment]
Curious if you're still having this issue. The bundled HTML5 parser may have been to blame since it's rather old. Which version of Dompdf are you using? We just released 0.8.4 which addresses some compatibility issues.
